pkg/log/log_file_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main under directory cmd parses and validates user input,
+// instantiates and initializes objects imported from pkg, and runs
+// the process.
+package log
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/component-base/logs"
+	"k8s.io/klog"
+)
+
+var (
+	testFlags          = initFlags()
+	klogDefaultMaxSize = klog.MaxSize
+)
+
+func initFlags() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	AddFlags(flags)
+	flags.AddGoFlagSet(flag.CommandLine)
+	return flags
+}
+
+func restoreFlagDefaultValues() {
+	testFlags.Set(logToStdErrFlag, "true")
+	testFlags.Set(logFileFlag, "")
+	testFlags.Set(logDirFlag, "")
+	testFlags.Set(maxSizeFlag, fmt.Sprintf("%d", klogDefaultMaxSize/1024/1024))
+	testFlags.Set(maxNumFlag, "0")
+
+	klog.MaxSize = klogDefaultMaxSize
+	logFileMaxNum = 0
+	logDir = ""
+}
+
+// Log ~100K bytes.
+func testLogging() {
+	// ~1K bytes per line.
+	line := make([]byte, 1000)
+	for i := 0; i < 1000; i++ {
+		line[i] = byte('0' + i%10)
+	}
+	// Log 100 lines, ~100K bytes.
+	for i := 0; i < 100; i++ {
+		klog.Infof("%d: %s", i, string(line))
+		klog.Warningf("%d: %s", i, string(line))
+	}
+	logs.FlushLogs()
+}
+
+func TestKlogFileLimits(t *testing.T) {
+	testLogDir, err := ioutil.TempDir("", "antrea-log-test")
+	if err != nil {
+		t.Errorf("Failed to create tmp log dir: %v", err)
+		return
+	}
+	defer os.RemoveAll(testLogDir)
+
+	args := []string{"--logtostderr=false", "--log_dir=" + testLogDir, "--log_file_max_size=1", "--log_file_max_num=2"}
+	testFlags.Parse(args)
+	InitLogFileLimits(testFlags)
+	logs.InitLogs()
+	defer restoreFlagDefaultValues()
+
+	// Should generate 3 log files (100K * 30 / 1M).
```

Comment:
maybe use `const`s for 100k, 1M and number of iterations

then we can add an assert like this to document our intent:
```
require.Greater(logSize * numIters / logFileMaxSize, logFileMaxNum, "test should generate enough logs to exceed --log_file_max_num")
```

Reply:
The problem is hard to accurately control the log message size/number, due to the log file/line header and klog flush time (I am not very sure about this but I did see log interval impact the number of log files created). So, I feel such assert does not help much.

Reply:
Sounds good. You probably want to include that information in the comment though.

Reply:
Ok. Added some comments to explain this.